src/Riok.Mapperly/Descriptors/MappingBuilders/InlineExpressionMappingBuilder.cs

```diff
@@ -25,21 +25,51 @@ public static class InlineExpressionMappingBuilder
         }
 
         var methodSyntax = methodSyntaxRef.GetSyntax();
-        if (methodSyntax is not MethodDeclarationSyntax { ExpressionBody: { } body, ParameterList.Parameters: [var sourceParameter] })
+
+        var bodyExpression = methodSyntax switch
```

Comment:
This adds quite a bit of complexity to this method, I think the readability could be improved a lot by extracting this into another method.